libs/math/tests/math/all_tests/serializers/fixed_point.cpp

```diff
@@ -41,16 +43,21 @@ TEST(FixedPointSerialisationTest, IntegerSerialisation)
 
 TEST(FixedPointSerialisationTest, DecimalSerialisation)
 {
-  for (float i(-10); i < 10; i += 0.12345f)
+  std::random_device               random_device;
```

Comment:
Out of curiosity, why not just change the for loop to a while:
`
float i = -10;
while(i < 10.f)
{
// ...
i += 0.12345f;
}
`
It would be a more natural fix to the problem as it would not change the logic of the test.

Reply:
Because that would have the same precision issues as the for loop, it's just that the checker may not be able to catch it...